compiler/noirc_evaluator/src/ssa/opt/mem2reg.rs

```diff
@@ -140,6 +140,9 @@ struct PerFunctionContext<'f> {
     /// Track whether a reference was passed into another entry point
     /// This is needed to determine whether we can remove a store.
     calls_reference_input: HashSet<ValueId>,
+    /// Track whether a reference was passed into a make array instruction
+    /// This is needed to determine whether we can remove a store.
+    make_array_references: HashSet<ValueId>,
```

Comment:
Does this need to be separate from `calls_reference_input`? Maybe they should be combined into `instructions_referencing_input`

Reply:
I felt it was clearer to also leave them separate, but I can merge and update the comments if that is preferred.